snort3: new package

MikePetullo opened the pull request:
Signed-off-by: W. Michael Petullo <mike@flyn.org>

Maintainer: me
Compile tested: x86_64 master
Run tested: x86_64

Description:
Snort 3.0.0-beta represents a more modern implementation of Snort. The folks behind Snort have not yet published a release schedule, and so I packages the beta version. This requires pull request #8490, which provides an updated libdaq.

MikePetullo commented:
See also http://lists.infradead.org/pipermail/openwrt-devel/2019-March/016388.html.

diizzyy commented on net/snort/Makefile:
```diff
@@ -8,29 +8,38 @@
   SECTION:=net
   CATEGORY:=Network
-  DEPENDS:=+libdaq +libdnet +libnghttp2 +libopenssl +libpcap +libpcre +libpthread +libuuid +zlib +SNORT_LZMA:liblzma
+  DEPENDS:=+libdaq +libdnet +libnghttp2 +libopenssl +libpcap +libpcre +libpthread +libuuid +zlib +libhwloc +luajit +libstdcpp +liblzma +libtirpc
```
Are libnghttp2, liblzma and libtirpc needed?

MikePetullo commented:
I pushed a new revision which instead introduces snort 3.0.0-beta as a new package, snort3. This allows OpenWrt to continue to providing libdaq and snort (2) while also providing libdaq22 and snort3. In addition, I addressed the comments from @diizzyy.

The snort3 package requires libtirpc, but I removed the references to the unnecessary libnghttp2 and liblzma.

diizzyy commented on net/snort3/Makefile:
```diff
@@ -0,0 +1,97 @@
+#
+# Copyright (C) 2006-2015 OpenWrt.org
```
Remove or add your own name

diizzyy commented on net/snort3/Makefile:
```diff
@@ -0,0 +1,97 @@
+	-D MAKE_DOC:BOOL=off \
+	-D MAKE_HTML:BOOL=off \
+	-D MAKE_PDF_DOC:BOOL=off \
+	-D MAKE_TEXT_DOC:BOOL=off \
```
Drop space between ```-D``` and MAKE...
For the sake of consistency please use caps

diizzyy commented on net/snort3/files/snort.init:
```diff
@@ -0,0 +1,42 @@
+#!/bin/sh /etc/rc.common
+# Copyright (C) 2015 OpenWrt.org
```
Drop this

diizzyy commented on net/snort3/files/snort.init:
```diff
@@ -0,0 +1,42 @@
+#!/bin/sh /etc/rc.common
+# Copyright (C) 2015 OpenWrt.org
+
+START=90
```
Any reason why 90 is set? Please use 99 otherwise

MikePetullo commented:
Addressed more comments from @diizzyy. Thank you!

MikePetullo commented:
Closed. We submit a merge request for both snort and snort3.
